processor/signozspanmetricsprocessor/config.go

```diff
@@ -80,6 +80,9 @@ type Config struct {
 
 	// skipSanitizeLabel if enabled, labels that start with _ are not sanitized
 	skipSanitizeLabel bool
+
+	// MetricsEmitInterval is the time period between when metrics are flushed or emitted to the configured MetricsExporter.
+	MetricsFlushInterval time.Duration `mapstructure:"metrics_flush_interval"`
```

Comment:
This interval is used to flush the delta metrics being calculated in-memory? I remember the initial implementation I had caused increase in writes to clickhouse as the calculated metrics were designed to be written immediately causing heavy DB write/s. I hope we should not face that with this configuration.

Another thing, this flush interval was there before the introduction of  delta temporality too? I want to understand why was it there before if we used to scrape things like prometheus and not push at flushInterval like defined here

Reply:
>This interval is used to flush the delta metrics being calculated in-memory?

Yes

> I remember the initial implementation I had caused increase in writes to clickhouse as the calculated metrics were designed to be written immediately causing heavy DB write/s. I hope we should not face that with this configuration.

Yes, this accumulates the values for each interval and then sends the data to the exporter.

> this flush interval was there before the introduction of delta temporality too?

No, it wasn't there. The previous design had the cumulative value which never resets but the Prometheus was configured to read the cumulative value every 1 minutes (or scrape interval).

Reply:
got it thanks.